src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java

```diff
@@ -24,12 +24,24 @@
 
     private final Context mContext;
 
+    private boolean mGenerate;
+
     private String mCommand;
 
     /**
      * Constructs instance.
      *
-     * @param command The exact command to be executed on the shell.
+     * @param generate Whether to generate keys, a config file and immediately exit.
+     */
+    public SyncthingRunnable(Context context, boolean generate) {
```

Comment:
Maybe it would make more sense to pass an `enum command`. Then you could do a switch here and set `mCommand` to the correct one. 

Enum values would be something like generate, run, test.